### PR TITLE
[ACR] `az acr check-health`: Fix "DOCKER_PULL_ERROR" on Windows

### PR DESCRIPTION
**Pull request by Wwwsylvia**
**Description**  

Fix [az acr check-health : Error image operating system "linux" cannot be used on this platform](https://github.com/Azure/acr/issues/328)
Fix the bug that `az acr check-health` always throws "DOCKER_PULL_ERROR" on Windows 

**Before:**
image

**After:**
image

**Testing Guide**  

`az acr check-health --name myregistry --ignore-errors --yes`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by Wwwsylvia** [comment]
@yugangw-msft Could you help to review?

**Comment by yonzhan** [comment]
add to S169

File: src/azure-cli/azure/cli/command_modules/acr/check_health.py [hunk 1]
```diff
@@ -108,7 +111,11 @@ def _get_docker_status_and_version(ignore_errors, yes):
         output, warning, stderr = _subprocess_communicate([docker_command, "pull", IMAGE])
 
         if stderr:
-            _handle_error(DOCKER_PULL_ERROR.append_error_message(stderr), ignore_errors)
+            if DOCKER_PULL_WRONG_PLATFORM in stderr:
+                print_pass("Docker pull of '{}'".format(IMAGE))
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/acr/check_health.py:115]: in cli we don't use print but logger, just checked this print_pass is used several places in this file. what scenario it's used for besides normal logger methods, could you pls change to logger.xxx?

Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/acr/check_health.py:115]: For `print_pass`, it's redirected to std_err. I think it should be fine.
https://github.com/Azure/azure-cli/blob/f5cbbb28b388ad1dd114ffaedea26446fe5acc8c/src/azure-cli/azure/cli/command_modules/acr/check_health.py#L37

Other information like versions is also printed in stderr, how do users use the result of `acr check-health` command in a script?

**Comment by qwordy** [comment]
How about output error without modification?
This line is not very robust. Usually, we should rely on error code rather than string matching.
```
if DOCKER_PULL_WRONG_PLATFORM in stderr:
```

File: src/azure-cli/azure/cli/command_modules/acr/check_health.py (hunk 1 above)
Comment (Wwwsylvia) [reply src/azure-cli/azure/cli/command_modules/acr/check_health.py:114]: Thanks, I've replaced print() with logger.warning().

> Other information like versions is also printed in stderr, how do users use the result of `acr check-health` command in a script?

`az acr check-health` is for troubleshooting purpose, usually users would not need the output of this command in script. A exit code is enough to check the status.

**Comment by Wwwsylvia** [comment]
> How about output error without modification?
> This line is not very robust. Usually, we should rely on error code rather than string matching.
> 
> ```
> if DOCKER_PULL_WRONG_PLATFORM in stderr:
> ```

We knew string matching is definitely not a good practice, but unfortunately it seems to be the only way to implement this, since currently Docker CLI does not return different exit codes to distinguish different errors types.

I found this [[Proposal] docker pull return code](https://github.com/docker/cli/issues/538) issue in their repo is still open...

**Comment by Wwwsylvia** [comment]
This PR is ready for review

File: src/azure-cli/azure/cli/command_modules/acr/check_health.py [hunk 2]
```diff
@@ -27,13 +26,14 @@
...
 NOTARY_VERSION_REGEX = re.compile(r'Version:\s+([.\d]+)')
+DOCKER_PULL_WRONG_PLATFORM = 'cannot be used on this platform'
 
 
 # Utilities functions
 def print_pass(message):
     from colorama import Fore, Style, init
     init()
-    print(str(message) + " : " + Fore.GREEN + "OK" + Style.RESET_ALL, file=sys.stderr)
+    logger.warning(str(message) + " : " + Fore.GREEN + "OK" + Style.RESET_ALL)
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/acr/check_health.py:36]: pls be aware of that warning will display to stderr. Some customers who're doing automation don't want warning in stderr to break auto pipeline

Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/acr/check_health.py:36]: and we don't suggest color in output, except it's from CLI underlying framework.

Comment (Wwwsylvia) [reply src/azure-cli/azure/cli/command_modules/acr/check_health.py]: Removed

**Comment by Wwwsylvia** [comment]
Can anyone help to merge if no more concerns?

**Pull request merged**